Donation-platform/src/main/webapp/personalizationScript.js

```diff
@@ -107,9 +139,211 @@ function displayTags(tags) {
 $(function() {
   $("#sortable").sortable();
   $('#submit').click(function() {
+
+    // Remove the first element in tagOrder (which is the instruction label)
+    // in order to get the tags.
+    var tagOrder = $("#sortable").sortable('toArray');
+    var tags = [];
+    for(i = 1; i < tagOrder.length; i++) {
+      tags.push(tagOrder[i]);
+    }
+
+    // If there are 1-3 tags in the rank, run loadPersonalizedCharities.
+    if(tags.length > 0 && tags.length <= 3) {
+      loadPersonalizedCharities(tags);
+    // Otherwise, if there are no tags in the rank, alert the user to select
+    // 3 causes before re-submitting (although 1 or 2 tags is also allowed, 
+    // encourage 3 so the user gets the most comprehensive results).
+    } else if(tags.length == 0) {
+      alert("Please select 3 causes to get personalized charities.");
+    // Otherwise, if there more than 3 tags in the rank, alert the user to select
+    // no more than 3 causes before re-submitting (NOTE: the function that places 
+    // tags into the ranking already prevents more than 3 tags from being added; 
+    // this is a fail-safe).
+    } else if(tags.length > 3) {
+      alert("Please select no more than 3 tags to get personalized charities");
```

Comment:
If you want to look into stylizing your alerts i can recommend sweetalert2 if you're interested. That's what I used in the addCharity.html page.